### DISCUSSION TITLE
Trouble closing popover from clicking trigger

### DISCUSSION THREAD

**Discussion by gogocharli** [adobe/react-spectrum#5452] [General]
I've referenced the documentation for the `usePopover` hook and I've started writing a custom component with the help from other helpers in the library.

However, I'm running into an issue when trying to close the popover from clicking the trigger after it has been open.

I have a [reproduction available here](https://codesandbox.io/s/magical-wilson-3vg3fh?file=/src/Popover.component.tsx).

I'm sure I'm missing something, but any help is appreciated!

**Comment by snowystinger** [comment]
A few things were switched around here's a fixed up version https://codesandbox.io/s/great-firefly-c647td?file=/src/Popover.component.tsx
You needed to useButton in your button most importantly. There were also some props and objects which weren't being passed to the correct place or Components rendered in an odd spot.

Otherwise, I'd just check how closely you match the example setup https://react-spectrum.adobe.com/react-aria/usePopover.html#example

You may find it easier though to use our new React Aria Components https://react-spectrum.adobe.com/react-aria/Popover.html they handle all the internal glue for you.

**Comment by gogocharli** [comment]
@snowystinger Thank you so much for looking into it and even fixing it up. That's very appreciated!

I'll also take a look at the Aria components as per your recommendation.
